applications/serial_lte_modem/src/slm_at_mqtt.c

```diff
@@ -563,17 +583,29 @@ static int handle_at_mqtt_connect(enum at_cmd_type cmd_type)
 			}
 			ctx.cid[cid_sz] = '\0';
 			err = at_params_string_get(&at_param_list, 3,
+						ctx.uname, &username_sz);
+			if (err < 0) {
+				return err;
+			}
+			ctx.uname[username_sz] = '\0';
+			err = at_params_string_get(&at_param_list, 4,
+						ctx.pword, &password_sz);
+			if (err < 0) {
+				return err;
+			}
+			ctx.pword[password_sz] = '\0';
+			err = at_params_string_get(&at_param_list, 5,
 							ctx.url, &url_sz);
 			if (err < 0) {
 				return err;
 			}
 			ctx.url[url_sz] = '\0';
-			err = at_params_int_get(&at_param_list, 4, &ctx.port);
+			err = at_params_int_get(&at_param_list, 6, &ctx.port);
 			if (err < 0) {
 				return err;
 			}
-			if (at_params_valid_count_get(&at_param_list) == 6) {
-				err = at_params_int_get(&at_param_list, 5,
+			if (at_params_valid_count_get(&at_param_list) == 8) {
```

Comment:
Since the parameter count increased, the initial check at line 570 should be updated.

Reply:
Thanks for the notification. The initial count is updated.